src/field/field3d.cxx

```diff
@@ -279,10 +279,10 @@ Field3D & Field3D::operator=(const Field3D &rhs) {
 
 Field3D & Field3D::operator=(const Field2D &rhs) {
   TRACE("Field3D = Field2D");
-  
-  /// Check that the data is valid
-  checkData(rhs);
- 
+
+  /// Check that the data is allocated
+  ASSERT1(rhs.isAllocated());
```

Comment:
Instead of `ASSERT` here we could just choose to return the unallocated `Field3D` so that we allow assignment from unallocated fields -- I'm not sure that would be good but is an option.